src/main/java/com/conveyal/datatools/manager/gtfsplus/GtfsPlusValidation.java

```diff
@@ -133,21 +137,34 @@ private static void validateTable(
             int index = fieldList.indexOf(fieldName);
             if (index != -1) {
                 // Add spec field for each field found.
-                fieldsFounds[index] = specField;
+                fieldsFound[index] = specField;
             } else if (isRequired(specField)) {
                 // If spec field not found, check that missing field was not required.
                 issues.add(new ValidationIssue(tableId, fieldName, -1, "Required column missing."));
             }
         }
         // Iterate over each row and validate each field value.
         int rowIndex = 0;
+        int rowsWithWrongNumberOfColumns = 0;
         while ((line = in.readLine()) != null) {
             String[] values = line.split(Consts.COLUMN_SPLIT, -1);
-            for (int v = 0; v < values.length; v++) {
-                validateTableValue(issues, tableId, rowIndex, values[v], fieldsFounds[v], gtfsFeed);
+            // First, check that row has the correct number of fields.
+            if (values.length != fieldsFound.length) {
+                rowsWithWrongNumberOfColumns++;
+            }
+            // Validate each value in row. Note: we iterate over the fields and not values because a row may be missing
+            // columns, but we still want to validate that missing value (e.g., if it is missing a required field).
+            for (int f = 0; f < fieldsFound.length; f++) {
+                // If value exists for index, use that. Otherwise, default to null to avoid out of bounds exception.
+                String val = f < values.length ? values[f] : null;
+                validateTableValue(issues, tableId, rowIndex, val, fieldsFound[f], gtfsFeed);
             }
             rowIndex++;
         }
+        // Add issue for wrong number of columns after processing all rows.
+        if (rowsWithWrongNumberOfColumns > 0) {
+            issues.add(new ValidationIssue(tableId, null, -1, rowsWithWrongNumberOfColumns + " row(s) do not contain the same number of fields as there are headers. (File may need to be edited manually.)"));
+        }
```

Comment:
I think it'd be better to create an issue for each row that has the incorrect amount of fields instead of just one per table.

Reply:
Yes, I considered this, but the reason I opted for this approach was that there's no concept of a row-level issue in the UI right now. So we would potentially need to add that to the UI somewhere. Also, there's the trouble of reporting the issue at the row level, but not really giving the user a great way to resolve the issue in the GTFS+ editor. Essentially, all of the rows with the wrong number of columns can be resolved simply by clicking the "Save and Revalidate" button -- so the resolution is more at the table level than the row level (like, for example, a bad value for a field would be). Not sure if all of that makes sense...

Reply:
Hmm, ok, sounds reasonable.

Reply:
I'll copy some version of this conversation into the code to avoid confusing someone later on.